specs/iota-rs-ENGINEERING-SPEC-0000.md

```diff
@@ -49,90 +50,89 @@ The data structure to initialize the instance of the Higher level client library
-| **value**          | ✘        | 0         | u64           | The amount of IOTA to send. If the value is zero the message object will have a IndexationPayload instead of a SignedTransactionPayload with an embedded IndexationPayload |
-| **path**           | ✘        | `m/0'/0'` | [BIP32Path]   | The wallet chain BIP32 path we want to search for.                                                                                                                         |
-| **output**         | ✘        | None      | \[Output\]    | Users can manually pick their own output instead of having node decide on which output should be used.                                                                     |
-| **indexation_key** | ✘        | None      | String        | An optional indexation key of the indexation payload.                                                                                                                      |
-| **data**           | ✘        | None      | [u8]          | An optional indexation data of the indexation payload.                                                                                                                     |
-
-* If only `indexation_key` and `data` are provided. This method will create a message with only indexation payload instead.
+| Parameter | Required | Default | Type | Definition |
+| - | - | - | - | - |
+| **seed** | ✘ | None | [Seed] | The seed of the account we are going to spend, only needed for transactions |
+| **account_index** | ✘ | 0 | usize | The account index, responsible for the value `✘` in the Bip32Path `m/44'/4218'/✘'/0'/0'`. |
+| **initial_address_index** | ✘ | 0 | usize | The index from where to start looking for balance. Responsible for the value `✘` in the Bip32Path `m/44'/4218'/0'/0'/✘'`. |
+| **input** | ✘ | None | UTXOInput | Users can manually select their UTXOInputs instead of having automatically selected inputs. |
+| **output** | ✘ | None | address: &[Bech32Address],<br />amount: u64 | Address to send to and amount to send. Address needs to be Bech32 encoded. |
+| **output_hex** | ✘ | None | address: &str,<br />amount: u64 | Address to send to and amount to send. Address needs to be hex encoded. |
+| **index** | ✘ | None | &str | An optional indexation key for an indexation payload. 1-64 bytes long. |
+| **data** | ✘ | None | Vec<u8> | Optional data for the indexation payload. |
+| **parent** | ✘ | None | [MessageId] | An optional parent [MessageId] to be used as one parent. |
+| **network_id** | ✘ | None | u64 | Optional network id, if not set it will be used from the nodeinfo. |
+
+Depending on the provided values this function will create a message with:
+
+* no payload
+* an indexation payload
+* a transaction payload
+* a transaction payload containing an indexation payload
 
 ### Return
 
 The [Message] object we build.
```

Comment:
Maybe here we need to return the `MessageId`, not `Message`? The Message can be get from the MessageId by `get_message().data()`, but we cannot get the `MessageId` from `Message`

Reply:
In the code we actually return the MessageId, but if you have the message you can also get the messageId with `.id()`

Reply:
Cool, saw that. Then we need to modify the code to return the `Message`.